dev/io.openliberty.mcp.internal_fat/fat/src/io/openliberty/mcp/internal/fat/lifecycle/tests/BeanLifecycleTest.java

```diff
@@ -0,0 +1,116 @@
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.skyscreamer.jsonassert.JSONAssert;
+
+import com.ibm.websphere.simplicity.ShrinkHelper;
+import com.ibm.websphere.simplicity.log.Log;
+
+import componenttest.annotation.Server;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import io.openliberty.mcp.internal.fat.lifecycle.tools.ClassTool;
+import io.openliberty.mcp.internal.fat.utils.HttpTestUtils;
+
+/**
+ *
+ */
+@SuppressWarnings("unused")
+@RunWith(FATRunner.class)
+public class BeanLifecycleTest {
+
+    @Server("mcp-server")
+    public static LibertyServer server;
+
+    public static void info(Class<?> clazz, String method, String message) {
+        Log.info(clazz, method, message);
+    }
+
+    @BeforeClass
+    public static void setup() throws Exception {
+        WebArchive war = ShrinkWrap.create(WebArchive.class, "beanLifecycleTest.war")
+                                   .addPackage(ClassTool.class.getPackage());
+
+        ShrinkHelper.exportDropinAppToServer(server, war, SERVER_ONLY);
+
+        server.startServer();
+    }
+
+    @AfterClass
+    public static void teardown() throws Exception {
+        server.stopServer();
+    }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testDependentBeanLifecycle() throws Exception {
+        server.setMarkToEndOfLog();
+        String request = """
+                        {
+                          "jsonrpc": "2.0",
+                          "id": "2",
+                          "method": "tools/call",
+                          "params": {
+                            "beanClass": "io.openliberty.mcp.internal_fat.lifecycle.tools.ClassTool",
+                            "name": "sayHello",
+                            "arguments": {
+                              "name": "World"
+                            }
+                          }
+                        }
+                        """;
+
+        String response = HttpTestUtils.callMCP(server, "/beanLifecycleTest", request);
+        Log.info(getClass(), "testDependentBeanLifecycle", "Raw MCP response: " + response);
+
+        // Strict Mode tests
+        String expectedResponseString = """
+                        {
+                          "id": "2",
+                          "jsonrpc": "2.0",
+                          "result": {
+                            "content": [
+                              { "type": "text", "text": "Hello, World" }
+                            ],
+                            "isError": false
+                          }
+                        }
+                        """;
+        JSONAssert.assertEquals(expectedResponseString, response, true);
+
+        server.waitForStringInLog("\\[LIFECYCLE] @PreDestroy ClassTool");
+        // Fetch all lifecycle-related log messages
+        List<String> lifecycleMessages = server.findStringsInLogs(".*\\[(LIFECYCLE|LOGGED)].*");
```

Comment:
```suggestion
        List<String> lifecycleMessages = server.findStringsInLogsUsingMark(".*\\[(LIFECYCLE|LOGGED)].*", server.getDefaultLogFile());
```

We should use the version of this method which uses the mark. Doesn't make much difference now, but it would if you added another similar test to this class.